src/utils.jl

```diff
@@ -0,0 +1,126 @@
+function _prepare_cohort_directory(
+    path::String
+)
+    mkpath(path)
+    return path
+end
+
+function _load_metadata(
+    metadata_path::String,
+    metadata_check::Bool
+)
+    if metadata_check && isfile(metadata_path)
+        return JSON3.read(read(metadata_path, String), Dict{String, Any})
+    end
+    return Dict{String, Any}()
+end
+
+function _write_metadata_entry(
+    metadata::Dict{String, Any},
+    id::Int,
+    version::Int,
+    last_modified::String
+)
+    metadata[string(id)] = Dict(
+        "id" => id,
+        "version" => version,
+        "lastModified" => last_modified,
+        "downloadedAt" => string(Dates.now())
+    )
+end
+
+function _save_metadata(
+    metadata::Dict{String, Any},
+    metadata_path::String
+)
+    open(metadata_path, "w") do file
+        JSON3.pretty(file, metadata, JSON3.AlignmentContext(indent=4))
+    end
+end
+
+function _save_cohort_json(
+    id::Int,
+    body,
+    save_dir::String
+)
+    path = joinpath(save_dir, "$(id).json")
+    open(path, "w") do file
+        JSON3.pretty(file, body, JSON3.AlignmentContext(indent=4))
+    end
+    return path
+end
+
+function _format_timestamp(
+    ms::Int
+)
+    date = Dates.unix2datetime(ms ÷ 1000)
+    return Dates.format(date, "yyyy-mm-ddTHH:MM:SS")
+end
```

Comment:
Can you actually eliminate this function and put this code directly in the definition function?